src/content/deployment/helm/_index.en.md

```diff
@@ -4,4 +4,123 @@ date: 2020-02-19T21:25:43+01:00
+
+```bash
+export KUBECONFIG=<kubeconfig-of-broker>
+helm repo add submariner-latest https://submariner-io.github.io/submariner-charts/charts
+kubectl -n kube-system create serviceaccount tiller
+kubectl create clusterrolebinding tiller --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
+helm init --service-account tiller
+kubectl -n kube-system rollout status deploy/tiller-deploy
+```
+
+### Exporting environment variables needed later
+
+```bash
+export BROKER_NS=submariner-k8s-broker
+export SUBMARINER_PSK=$(LC_CTYPE=C tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 64 | head -n 1)
+export SUBMARINER_NS=submariner
+```
+
+### Deploying the broker
+
+```bash
+helm install submariner-latest/submariner-k8s-broker \
+             --name "${BROKER_NS}" \
+             --namespace "${BROKER_NS}" \
+             --set submariner.serviceDiscovery=true
+```
+
+Setup more environment variables we will need later for joining clusters.
+
+```bash
+export SUBMARINER_BROKER_URL=$(kubectl -n default get endpoints kubernetes \
+    -o jsonpath="{.subsets[0].addresses[0].ip}:{.subsets[0].ports[?(@.name=='https')].port}")
+export SUBMARINER_BROKER_CA=$(kubectl -n "${BROKER_NS}" get secrets \
+    -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_NS}-client')].data['ca\.crt']}")
+export SUBMARINER_BROKER_TOKEN=$(kubectl -n "${BROKER_NS}" get secrets \
+    -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_NS}-client')].data.token}" \
+       | base64 --decode)
+```
+
+### Joining a cluster
+
+This steps need to be repeated for every cluster you want to connect with Submariner.
+
+```bash
+export KUBECONFIG=kubeconfig-of-the-cluster-to-join
+export CLUSTER_ID=the-id-of-the-cluster
+export CLUSTER_CIDR=x.x.x.x/x   # the cluster's pod IP CIDR
+export SERVICE_CIDR=x.x.x.x/x   # the cluster's service IP CIDR
+```
+
+If your clusters have overlapping IPs (Pods/ClusterIPs), please set:
+
+```bash
+export GLOBALNET=true
+export GLOBAL_CIDR=169.254.x.x/x # using an individual non-overlapping
+                                 # range for each cluster you join.
+```
+
+Joining the cluster:
+
+```bash
+helm install submariner-latest/submariner \
+        --name submariner \
+        --namespace "${SUBMARINER_NS}" \
+        --set ipsec.psk="${SUBMARINER_PSK}" \
+        --set broker.server="${SUBMARINER_BROKER_URL}" \
+        --set broker.token="${SUBMARINER_BROKER_TOKEN}" \
+        --set broker.namespace="${BROKER_NS}" \
+        --set broker.ca="${SUBMARINER_BROKER_CA}" \
+        --set submariner.clusterId="${CLUSTER_ID}" \
+        --set submariner.clusterCidr="${CLUSTER_CIDR}" \
+        --set submariner.serviceCidr="${SERVICE_CIDR}" \
+        --set submariner.globalCidr="${GLOBAL_CIDR}" \
+        --set serviceAccounts.globalnet.create="${GLOBALNET}" \
+        --set submariner.natEnabled="true" \  # disable this if no NAT/floating IP will happen between gateways
```

Comment:
Not sure if I understand the floating IP here.. maybe better to drop it?

Reply:
Done